awscli/customizations/s3/fileinfo.py

```diff
@@ -216,7 +216,10 @@ def _handle_object_params(self, params):
                                      'permission=principal')
                 params[self._permission_to_param(permission)] = grantee
         if self.parameters['sse']:
-            params['ServerSideEncryption'] = 'AES256'
+            sse = self.parameters['sse'][0]
+            if sse == 'aws:kms':
+                self.client._request_signer._signature_version = 's3v4'
```

Comment:
Directly changing the `RequestSigner`'s `_signature_version` is pretty hacky but at this point in the code, I'm not able to attach to the `event_emitter` (AFAIK) to mutate it.